linetools/analysis/absline.py

```diff
@@ -284,3 +287,101 @@ def sum_logN(obj1, obj2):
             pass # Take sums
     # Return
     return flag_N, logN, sig_logN
+
+
+def get_tau0(wa0, fosc, logN, b):
+    """Get the value of the optical depth at the line center,
+    tau0. Taken from Draine 2011 (see Chapter 9). It neglects stimulated
+    emission which is fine for IGM or ISM except for radio-frequency
+    transitions.
+
+    Parameters
+    ----------
+    wa0 : Quantity
+        Rest-frame wavelength of the transition
+    fosc : float
+        Oscillator strength of the transition
+    logN : float, or np.array
+        log10 of the column density in cm^{-2}
+    b : Quantity, or Quantity array (same shape as logN)
+        Doppler parameter
+
+    Returns
+    -------
+    tau0: float or array
+        Optical depth at the line center. If logN and b are
+        arrays they must be of same shape.
+    """
+    # check format for logN and b
+    try:
+        n = len(logN)
+        if n != len(b):
+            raise IOError('If logN is array, b must be array of same shape.')
+    except TypeError:  # assuming float for logN
+        if not isinstance(logN, (float, int)):
+            raise IOError('If logN is not array, it must be float.')
+    # force to numpy array
+    logN = np.array(logN)
+
+    # convert units to CGS
+    b_cgs = b.to('cm/s').value
+    wa0_cgs = wa0.to('cm').value
+    N_cgs = 10**logN  # assumed to be in cm^2
+
+    # tau0
+    tau0 = np.sqrt(np.pi) * e2_me_c_cgs * N_cgs * fosc * wa0_cgs  / b_cgs  # eq. 9.8 Draine 2011
```

Comment:
I'd use units the way they prefer to be used.   That is, keep them all (instead of taking values) and do a `decompose()` on tau0 to insure it is unitless.  This means adding cm^-2 on N_cgs (or better, having it read in with units)